docs/reference.md

```diff
@@ -303,6 +311,71 @@ Specify the config file to write to. Without this option, your configuration
 will be written to `~/.gemstash/config.yml`. If you write to a custom location,
 you will need to pass the `--config-file` option to all Gemstash commands.
 
+## Preload
+
+Preload all the gems in your gemstash server from the default upstream.
+This can be useful if you plan to go somewhere where the internet connection
+will not be reliable or available.
+It is worth noting that this will require a large amount of disk space and
+computing time.
+
+### Usage
+
+```
+gemstash preload
+gemstash preload --latest
+gemstash preload --limit=1000
+gemstash preload --limit=1000 --skip=1000
+gemstash preload --threads=10 --server_url=http://my-server-url:9292
+```
+
+### Options
+
+#### --latest
+
+**Usage:** `--latest`
+
+**Description**<br />
+Only fetch the latest specs. This will limit the amount of disk space required,
+but there is no guarantee that the gems you are using will have all their
+dependencies met.
+
+#### --skip
+
+**Usage:** `--skip=<number>`
+
+**Description**<br />
+Number of gems to skip when preloading. This can be useful to load in chunks or
+to resume a gem preloading.
+
+#### --limit
+
+**Usage:** `--limit=<number>`
+
+**Description**<br />
+Limit for the number of gems to preload. In conjunction with [--skip](#--skip)
+this can be used for loading gems in chunks or resuming the process.
```

Comment:
Here's the scenario I'm thinking of that might be most potential for problems:

Lets say I have iterated over 10,000 gems. Let's say gem 10,001 happens to be an important one for me. Someone yanks a gem in the first 10,000, say gem 100. Gem 10,001 is now 10,000 in specs.4.8.gz, so when I run the next block, I start on what would have been gem 10,002.

Granted, only the gems on the borders of your limits are at risk, but given how long this takes, a yank becomes more likely. It also might not be a big problem most of the time, so maybe it's not worth worrying about.